### ISSUE TITLE
Python 3.12 changes in regex special characters

### ISSUE THREAD

**Issue by giachello** [nutechsoftware/alarmdecoder#73]
Getting the following error messages 

```
2024-04-08 08:58:20.108 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.12/site-packages/alarmdecoder/decoder.py:605: SyntaxWarning: invalid escape sequence '\.'
  matches = re.match('^!Sending(\.{1,5})done.*', data)

2024-04-08 08:58:20.143 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.12/site-packages/alarmdecoder/messages/panel_message.py:73: SyntaxWarning: invalid escape sequence '\['
  _regex = re.compile('^(!KPM:){0,1}(\[[a-fA-F0-9\-]+\]),([a-fA-F0-9]+),(\[[a-fA-F0-9]+\]),(".+")$')

2024-04-08 08:58:20.198 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.12/site-packages/alarmdecoder/zonetracking.py:186: SyntaxWarning: invalid escape sequence '\d'
  zone_regex = re.compile('^CHECK (\d+).*$')
```

This is caused by a change in Python 3.12 (see https://docs.python.org/dev/whatsnew/3.12.html#other-language-changes) 

Should be a pretty easy fix to add and r in front of strings containing escape characters.

**Comment by pleasantone** [comment]
Agreed. I wonder if we're going to have to fork this given the lack of response?

**Comment by giachello** [comment]
I emailed the author on linkedin and he said he'd approve a pull request fixing it. I would do that instead of forking.

**Comment by pleasantone** [comment]
He approved but never merged a previous pull request. If you're in contact with him, could you ask him to review the pull requests and merge those he agrees with? Otherwise we need to talk about getting new maintainers or forking.

**Comment by ajschmidt8** [comment]
I just merged https://github.com/nutechsoftware/alarmdecoder/pull/75.

Can this issue be closed now?

**Comment by pleasantone** [comment]
Yes, could you please tag and push to pypi? Thank you so much!

**Comment by ajschmidt8** [comment]
> Yes, could you please tag and push to pypi? Thank you so much!
> […](#)

No problem. I've completed those steps. More details here for awareness https://github.com/home-assistant/core/issues/133249#issuecomment-2676931740.

**Comment by ajschmidt8** [comment]
Also posted a comment here about my role with AlarmDecoder: https://github.com/nutechsoftware/alarmdecoder/issues/74#issuecomment-2676972657

**Comment by pleasantone** [comment]
Thank you SO MUCH!

**Comment by ajschmidt8** [comment]
No problem. Happy to help. Please be sure to close this issue once you consider it complete.

**Comment by giachello** [comment]
Thanks for taking care of this.
